### PR TITLE
Support recursive matching MultipleFileUpload with `glob=True`

### PR DESCRIPTION
**Pull request by angeloudy**
This will support `**` in python `glob.glob`.

By default `recursive` is set to `False`. Setting it to `True` wouldn't hurt anything.
The only disadvantage introduced by recursive matching is performance downgrade, which can be avoided by not using `**`.

### CODE REVIEW

**Comment by tardyp** [comment]
can you please create a newsfragment, and be clear that worker needs to be updated

**Comment by tardyp** [comment]
Almost done!

looks like recursive is not supported by python 2.7.

Can you please add a test for 2.7, and document that glob supports recursive only if the worker is on python3?

**Comment by tardyp** [comment]
Can you please look why unit tests are failing?

Also eventually, please squash your commit to cleanup history.
thanks

File: worker/buildbot_worker/commands/fs.py [hunk 1]
```diff
@@ -235,7 +235,11 @@ def start(self):
         pathname = os.path.join(self.builder.basedir, self.args['path'])
 
         try:
-            files = glob.glob(pathname)
+            # recursive matching is only support in python3.5+
+            if sys.version_info[:2] >= (3, 5):
+                files = glob.glob(pathname, recursive=True)
```
Comment (sa2ajj) [inline worker/buildbot_worker/commands/fs.py]: just saying (i'm not sure if it's easier to read): `files = glob.glob(pathname, recursive=True if sys.version_info[:2] >= (3, 5) else False)`

Comment (angeloudy) [reply worker/buildbot_worker/commands/fs.py]: The problem with your code is that python2 does not support `recursive` at all, you can't just pass `false` to it.

Comment (sa2ajj) [reply worker/buildbot_worker/commands/fs.py]: right, this is the build worker code, sorry

**Comment by tardyp** [comment]
Could you please squash your commits?

**Comment by angeloudy** [comment]
@tardyp Sorry, I don't use git quite often. What's squash? what command I use to do it?

**Comment by tardyp** [comment]
Hi squashing is delicate but important skill to have when contributing to open source.

https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

**Comment by angeloudy** [comment]
I tried the `rebase` command. There are lots of commits that are not mine. Are they from updating branch?

**Comment by angeloudy** [comment]
@tardyp I've squashed the commits. Can you merge it now?

**Comment by angeloudy** [comment]
@tardyp Can you help get this one merged?

**Comment by angeloudy** [comment]
Can someone merge this?

**Pull request merged**